src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumMiniWallet.kt

```diff
@@ -0,0 +1,188 @@
+package fr.acinq.lightning.blockchain.electrum
+
+import fr.acinq.bitcoin.*
+import fr.acinq.lightning.utils.Connection
+import fr.acinq.lightning.utils.lightningLogger
+import fr.acinq.lightning.utils.sat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlin.native.concurrent.ThreadLocal
+
+data class WalletState(val addresses: Map<String, List<UnspentItem>>, val privateKeys: Map<String, PrivateKey>, val parentTxs: Map<ByteVector32, Transaction>) {
+    val utxos: List<UnspentItem> = addresses.flatMap { it.value }
+    val balance: Satoshi = utxos.sumOf { it.value }.sat
+
+    data class Utxo(val previousTx: Transaction, val outputIndex: Int, val blockHeight: Long) {
+        val outPoint = OutPoint(previousTx, outputIndex.toLong())
+        val amount = previousTx.txOut[outputIndex].amount
+    }
+
+    fun spendable(): List<Utxo> {
+        return addresses
+            .filter { privateKeys.containsKey(it.key) }
+            .flatMap { it.value }
+            .filter { parentTxs.containsKey(it.tx_hash) }
```

Comment:
Sane here, isn't there a confusion between `txid` and `tx_hash`?

Reply:
Fixed in 6a66e25.